components/ai_chat/core/browser/engine/engine_consumer_conversation_api_unittest.cc

```diff
@@ -1009,12 +1230,193 @@ TEST_F(EngineConsumerConversationAPIUnitTest, GetStrArrFromResponse) {
+                                                      mojom::APIError> result) {
+          ASSERT_FALSE(result.has_value());
+          EXPECT_EQ(result.error(), mojom::APIError::InternalError);
+        }));
+
+    testing::Mock::VerifyAndClearExpectations(mock_api_client);
+  }
+
+  // Test null event
+  {
+    EXPECT_CALL(*mock_api_client, PerformRequest(_, _, _, _, _))
+        .WillOnce([&](const std::vector<ConversationEvent>& conversation,
+                      const std::string& language,
+                      EngineConsumer::GenerationDataCallback data_callback,
+                      EngineConsumer::GenerationCompletedCallback callback,
+                      const std::optional<std::string>& model_name) {
+          std::move(callback).Run(base::ok(
+              EngineConsumer::GenerationResultData(nullptr, std::nullopt)));
+        });
+
+    engine_->GenerateQuestionSuggestions(
+        is_video, page_content, selected_language,
+        base::BindLambdaForTesting([&](base::expected<std::vector<std::string>,
+                                                      mojom::APIError> result) {
+          ASSERT_FALSE(result.has_value());
+          EXPECT_EQ(result.error(), mojom::APIError::InternalError);
+        }));
+
+    testing::Mock::VerifyAndClearExpectations(mock_api_client);
+  }
+
+  // Test non-completion event
+  {
+    EXPECT_CALL(*mock_api_client, PerformRequest(_, _, _, _, _))
+        .WillOnce([&](const std::vector<ConversationEvent>& conversation,
+                      const std::string& language,
+                      EngineConsumer::GenerationDataCallback data_callback,
+                      EngineConsumer::GenerationCompletedCallback callback,
+                      const std::optional<std::string>& model_name) {
+          auto selected_language_event =
+              mojom::ConversationEntryEvent::NewSelectedLanguageEvent(
+                  mojom::SelectedLanguageEvent::New("en-us"));
+          std::move(callback).Run(base::ok(EngineConsumer::GenerationResultData(
+              std::move(selected_language_event), std::nullopt)));
+        });
+
+    engine_->GenerateQuestionSuggestions(
+        is_video, page_content, selected_language,
+        base::BindLambdaForTesting([&](base::expected<std::vector<std::string>,
+                                                      mojom::APIError> result) {
+          ASSERT_FALSE(result.has_value());
+          EXPECT_EQ(result.error(), mojom::APIError::InternalError);
+        }));
+
+    testing::Mock::VerifyAndClearExpectations(mock_api_client);
+  }
+}
+
+TEST_F(EngineConsumerConversationAPIUnitTest,
+       GenerateAssistantResponse_WithModelKeyOverride) {
+  auto* mock_api_client = GetMockConversationAPIClient();
+
+  // Expect PerformRequest with the overridden model name
+  EXPECT_CALL(
+      *mock_api_client,
+      PerformRequest(
+          _, _, _, _,
+          testing::Eq(std::optional<std::string>("llama-3-8b-instruct"))))
```

Comment:
we should be using `GetLeoModelNameByKey` here instead of hard coded the value. Verifying key mapping to the correct name belongs to the scope of ModelService unit test.

Reply:
https://github.com/brave/brave-core/pull/28999/commits/e3095a4c286abe8eec7d718a0e0fccca4507b51e